### PR TITLE
Lamil/responses

### PR DESCRIPTION
**Pull request by lauren-mills**
## Description

Updates to responses:
- Consolidated response logic into one ResponseManager class

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the appropriate unit tests
- [ ] I have tested any new/updated dialogs using Speech in the emulator to ensure the [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) property is set to enable a high quality speech-first experience and the appropriate [InputHints](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) are set correctly. 
- [ ] I have updated related documentation

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
- [ ] A duplicate issue is filed to track future  work.

If you have updated responses or `.lu` files:
- [x] All languages have been updated
- [x] You have tested deployment with your new models

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/Microsoft.Bot.Solutions/Microsoft.Bot.Solutions.csproj [hunk 1]
```diff
@@ -6,6 +6,42 @@
...
+
+  <ItemGroup>
+    <Compile Remove="Dialogs\BotResponseBuilder.cs" />
+    <Compile Remove="Dialogs\IBotResponseBuilder.cs" />
+    <Compile Remove="Dialogs\ResponseManager.cs" />
+    <Compile Remove="Extensions\TurnContextEx.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Remove="Resources\CommonResponses.de.json" />
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/Microsoft.Bot.Solutions/Microsoft.Bot.Solutions.csproj:26]: Are these needed?

**Review by darrenj — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Responses/ResponseManager.cs [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+                foreach (var locale in locales)
+                {
+                    try
+                    {
+                        resourceAssembly = resourceAssembly.GetSatelliteAssembly(new CultureInfo(locale));
+                        LoadResponses(resourceName, resourceAssembly, locale);
+                    }
+                    catch
+                    {
+                        // If satellite assembly doesn't exist, bot will fall back to default.
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Responses/ResponseManager.cs:43]: should log this so the developer can be notified when this error occurs.

**Review by lzc850612 — CHANGES_REQUESTED** [review]

**Comment by darrenj** [comment]
We should also add something to the create a skill docs to talk about response handling - the methods we provide, embeddedresource being needed, etc.

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Responses/ResponseManager.cs (hunk 2 above)
Comment (lauren-mills) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Responses/ResponseManager.cs:43]: Will resolve in future PR

**Pull request merged**